File: _knowledge_archive/scratch/audit.py
import os
import json

root_dir = r"d:\SLAMET\other\ai-agent-project"
ignore_dirs = {'node_modules', '.git', 'dist', 'build', '.svelte-kit', '.gemini', '.vscode', 'supabase\\.branches'}

def get_tree(startpath, max_depth):
    tree = {}
    for root, dirs, files in os.walk(startpath):
        dirs[:] = [d for d in dirs if d not in ignore_dirs]
        level = root.replace(startpath, '').count(os.sep)
        if level >= max_depth:
            del dirs[:]
        
        rel_path = os.path.relpath(root, startpath)
        if rel_path == '.':
            rel_path = 'ROOT'
        tree[rel_path] = {'dirs': dirs, 'files': len(files)}
    return tree

def count_loc(startpath):
    total_files = 0
    total_loc = 0
    loc_by_ext = {}
    
    for root, dirs, files in os.walk(startpath):
        dirs[:] = [d for d in dirs if d not in ignore_dirs]
        for f in files:
            ext = os.path.splitext(f)[1].lower()
            if ext in {'.ts', '.js', '.svelte', '.py', '.html', '.css', '.json'}:
                total_files += 1
                try:
                    with open(os.path.join(root, f), 'r', encoding='utf-8') as file:
                        lines = sum(1 for _ in file)
                        total_loc += lines
                        loc_by_ext[ext] = loc_by_ext.get(ext, 0) + lines
                except Exception:
                    pass
    return total_files, total_loc, loc_by_ext

tree = get_tree(root_dir, 3)
files, loc, loc_by_ext = count_loc(root_dir)

# Find bundle size if dist or build exists
bundle_size = 0
dist_path = os.path.join(root_dir, 'frontend', 'dist')
if os.path.exists(dist_path):
    for root, dirs, files in os.walk(dist_path):
        for f in files:
            bundle_size += os.path.getsize(os.path.join(root, f))
elif os.path.exists(os.path.join(root_dir, 'frontend', 'build')):
    for root, dirs, files in os.walk(os.path.join(root_dir, 'frontend', 'build')):
        for f in files:
            bundle_size += os.path.getsize(os.path.join(root, f))

result = {
    'tree': tree,
    'stats': {
        'total_files': files,
        'total_loc': loc,
        'loc_by_ext': loc_by_ext,
        'bundle_size_bytes': bundle_size
    }
}

with open(os.path.join(root_dir, 'scratch', 'audit_result.json'), 'w') as f:
    json.dump(result, f, indent=2)

print("Done")

File: _knowledge_archive/scratch/audit.js
const fs = require('fs');
const path = require('path');

const rootDir = "d:\\SLAMET\\other\\ai-agent-project";
const ignoreDirs = new Set(['node_modules', '.git', 'dist', 'build', '.svelte-kit', '.gemini', '.vscode', 'supabase\\.branches']);

function getTree(startPath, maxDepth) {
    const tree = {};
    
    function walk(currentPath, currentDepth) {
        if (currentDepth > maxDepth) return;
        
        let items = [];
        try {
            items = fs.readdirSync(currentPath, { withFileTypes: true });
        } catch (e) {
            return;
        }
        
        const dirs = [];
        let fileCount = 0;
        
        for (const item of items) {
            if (item.isDirectory()) {
                if (!ignoreDirs.has(item.name)) {
                    dirs.push(item.name);
                    walk(path.join(currentPath, item.name), currentDepth + 1);
                }
            } else {
                fileCount++;
            }
        }
        
        let relPath = path.relative(rootDir, currentPath);
        if (relPath === '') relPath = 'ROOT';
        
        // only store up to max depth in the tree structure
        if (currentDepth <= maxDepth) {
            tree[relPath] = { dirs, files: fileCount };
        }
    }
    
    walk(startPath, 0);
    return tree;
}

function countLoc(startPath) {
    let totalFiles = 0;
    let totalLoc = 0;
    const locByExt = {};
    
    function walk(currentPath) {
        let items = [];
        try {
            items = fs.readdirSync(currentPath, { withFileTypes: true });
        } catch (e) {
            return;
        }
        
        for (const item of items) {
            const fullPath = path.join(currentPath, item.name);
            if (item.isDirectory()) {
                if (!ignoreDirs.has(item.name)) {
                    walk(fullPath);
                }
            } else {
                const ext = path.extname(item.name).toLowerCase();
                if (['.ts', '.js', '.svelte', '.py', '.html', '.css', '.json'].includes(ext)) {
                    totalFiles++;
                    try {
                        const content = fs.readFileSync(fullPath, 'utf-8');
                        const lines = content.split('\n').length;
                        totalLoc += lines;
                        locByExt[ext] = (locByExt[ext] || 0) + lines;
                    } catch (e) {}
                }
            }
        }
    }
    
    walk(startPath);
    return { totalFiles, totalLoc, locByExt };
}

function getBundleSize() {
    let bundleSize = 0;
    
    function walk(currentPath) {
        let items = [];
        try {
            items = fs.readdirSync(currentPath, { withFileTypes: true });
        } catch (e) {
            return;
        }
        for (const item of items) {
            const fullPath = path.join(currentPath, item.name);
            if (item.isDirectory()) {
                walk(fullPath);
            } else {
                bundleSize += fs.statSync(fullPath).size;
            }
        }
    }
    
    const distPath = path.join(rootDir, 'frontend', 'dist');
    const buildPath = path.join(rootDir, 'frontend', 'build');
    
    if (fs.existsSync(distPath)) walk(distPath);
    else if (fs.existsSync(buildPath)) walk(buildPath);
    
    return bundleSize;
}

const tree = getTree(rootDir, 3);
const stats = countLoc(rootDir);
const bundleSize = getBundleSize();

stats.bundleSizeBytes = bundleSize;

const result = { tree, stats };

fs.writeFileSync(path.join(rootDir, 'scratch', 'audit_result.json'), JSON.stringify(result, null, 2));
console.log("Done");
